nautobot/extras/forms.py

```diff
@@ -227,33 +228,78 @@ def _append_relationships(self):
+                        association = queryset.first()
+                        if association:
+                            self.fields[field_name].initial = getattr(association, peer_side)
 
                 # Annotate the field in the list of Relationship form fields
                 self.relationships.append(field_name)
 
+    def clean(self):
+        """
+        Verify that any requested RelationshipAssociations do not violate relationship cardinality restrictions.
+
+        - For TYPE_ONE_TO_MANY and TYPE_ONE_TO_ONE relations, if the form's object is on the "source" side of
+          the relationship, verify that the requested "destination" object(s) do not already have any existing
+          RelationshipAssociation to a different source object.
+        - For TYPE_ONE_TO_ONE relations, if the form's object is on the "destination" side of the relationship,
+          verify that the requested "source" object does not have an existing RelationshipAssociation to
+          a different destination object.
+        """
+        for side, relationships in self.instance.get_relationships().items():
+            for relationship in relationships:
+                # If this side of the relationship is a "MANY" relation,
+                # we can coexist with other existing RelationshipAssociations and there's nothing to check now.
+                if relationship.has_many(side):
+                    continue
+
+                peer_side = RelationshipSideChoices.OPPOSITE[side]
+                field_name = f"cr_{relationship.slug}__{peer_side}"
+
+                # Is the form trying to set this field (create/update a RelationshipAssociation(s))?
+                # If not (that is, clearing the field / deleting RelationshipAssociation(s)), we don't need to check.
+                if field_name not in self.cleaned_data or not self.cleaned_data[field_name]:
+                    continue
+
+                # Are any of the objects we want a relationship with already entangled with another object?
+                if relationship.has_many(peer_side):
+                    target_peers = [item for item in self.cleaned_data[field_name]]
+                else:
+                    target_peers = [self.cleaned_data[field_name]]
+                for target_peer in target_peers:
+                    existing_peer_associations = RelationshipAssociation.objects.filter(
+                        relationship=relationship,
+                        **{
+                            f"{peer_side}_id": target_peer.pk,
+                        },
+                    ).exclude(**{f"{side}_id": self.instance.pk})
+                    if existing_peer_associations.count() > 0:
+                        raise ValidationError(
+                            {field_name: f"{target_peer} is already involved in a {relationship} relationship"}
+                        )
+
+        return super().clean()
```

Comment:
No need for a return value on `clean()`.
```suggestion
        super().clean()
```